coin_prices_from_coingecko/README.md

```diff
@@ -0,0 +1,283 @@
+  and set it in `fn-call.json` in the `api_key` field.
+
+## Quick Start
+
+To run this example, call:
+
+```bash
+make run
+```
+
+You will see the following output extracted from a Blocky AS response showing
+you the price of Bitcoin in USD on the Binance market:
+
+```json
+{
+  "Success": true,
+  "Error": "",
+  "Value": {
+    "market": "Binance",
+    "coin_id": "BTC",
+    "currency": "USD",
+    "price": 83799,
+    "timestamp": "2025-03-04T03:20:35Z"
+  }
+}
+```
+
+## Walkthrough
+
+Let's say you want to implement a simple price feed oracle that fetches the
+price of Bitcoin in USD. We can get the price from the CoinGecko API using their
+ticker API:
+
+```bash
+curl https://api.coingecko.com/api/v3/coins/bitcoin/tickers | jq .
+```
+
+If you run the above command, you will get a lot of information from multiple
+markets. Let's say you just want to get the price reported by Binance. In this
+example, you will write a Go function that fetches the ticker data, finds the
+Binance entry, and parses out the price.
+
+### Step 1: Create a parameterized oracle function
+
+We'll implement the oracle as `priceFunc` in
+[`main.go`](./main.go). As in previous examples, we will call this function
+using the `bky-as` CLI by passing in the
+[`fn-call.json`](./fn-call.json) file contents:
+
+```json
+[
+  {
+    "code_file": "tmp/x.wasm",
+    "function": "priceFunc",
+    "input": {
+      "market": "Binance",
+      "coin_id": "bitcoin"
+    },
+    "secret": {
+      "api_key": "CoinGecko API key"
+    }
+  }
+]
+```
+
+Notice the `input` section, which contains the parameters for `priceFunc`,
+specifically the `market` field set to "Binance" and the `coin_id` field set to
+"bitcoin". The `secret` section contains the `api_key` field, which you should
+set to your CoinGecko API key. Of course, you can change these values to get
+the price of other coins or from other markets.
+
+Next, we define the `priceFunc` function:
+
+```go
+type Args struct {
+	Market string `json:"market"`
+	CoinID string `json:"coin_id"`
+}
+
+type SecretArgs struct {
+	CoinGeckoAPIKey string `json:"api_key"`
+}
+
+//export priceFunc
+func priceFunc(inputPtr, secretPtr uint64) uint64 {
+	var input Args
+	inputData := as.Bytes(inputPtr)
+	err := json.Unmarshal(inputData, &input)
+	if err != nil {
+		outErr := fmt.Errorf("could not unmarshal input args: %w", err)
+		return WriteError(outErr)
+	}
+
+	var secret SecretArgs
+	secretData := as.Bytes(secretPtr)
+	err = json.Unmarshal(secretData, &secret)
+	if err != nil {
+		outErr := fmt.Errorf("could not unmarshal secret args: %w", err)
+		return WriteError(outErr)
+	}
+
+	price, err := getPriceFromCoinGecko(
+		input.Market,
+		input.CoinID,
+		secret.CoinGeckoAPIKey,
+	)
+	if err != nil {
+		outErr := fmt.Errorf("getting price: %w", err)
+		return WriteError(outErr)
+	}
+
+	return WriteOutput(price)
+}
+```
+
+The function takes two `uint64` arguments and returns a `uint64`. These are fat
```

Comment:
Consider adding a tutorial for processing input and secrets and instead of explaining it here, explain it in the input and secret input tutorial and just reference it in this one.

Reply:
I thought about it, but thought that it would be a bit too in a vacuum. We can certainly circle back to this idea though.

For now, I'm going to add a directory bullets to to the top level README, so that that people can at least find where to learn about input and output